### PR TITLE
docs: initial 0.4.0 breaking code migration docs

### PR DESCRIPTION
**Pull request by TimoGlastra**
Adds the 0.4.0 migration docs for breaking code only for now. Still need to do the breaking storage, but will do that in a follow up PR. 

This was definitely the hardest one to write from all updates we've had because there's a lot of breaking changes... and we haven't even done the indy to askar migration script docs (that will be a separate guilde), or the new setup.

### CODE REVIEW

File: guides/updating/versions/0.3-to-0.4.md [hunk 1]
```diff
@@ -0,0 +1,875 @@
...
+:::
+
+<!--tabs-->
+
+#### React Native
+
+First install the updated dependencies. Make sure to also install the new `@aries-framework/indy-sdk` package, which is a wrapper around the Indy SDK and install the `indy-sdk-react-native` package.
+
+```sh
+yarn add @aries-framework/react-native@^0.4.0 @aries-framework/core@^0.4.0 @aries-framework/indy-sdk@^0.4.0 @aries-framework/anoncreds@^0.4.0 indy-sdk-react-native@^0.4.0
```
Comment (karimStekelenburg) [inline guides/updating/versions/0.3-to-0.4.md:24]:
```suggestion
yarn add @aries-framework/react-native@^0.4.0 @aries-framework/core@^0.4.0 @aries-framework/indy-sdk@^0.4.0 @aries-framework/anoncreds@^0.4.0 indy-sdk-react-native@^0.3.1
```

File: guides/updating/versions/0.3-to-0.4.md [hunk 2]
```diff
@@ -0,0 +1,875 @@
...
+
+# or NPM
+npn install @aries-framework/react-native@^0.4.0 @aries-framework/core@^0.4.0 @aries-framework/indy-sdk@^0.4.0 @aries-framework/anoncreds@^0.4.0 indy-sdk-react-native@^0.4.0
```
Comment (karimStekelenburg) [inline guides/updating/versions/0.3-to-0.4.md:27]:
```suggestion
npn install @aries-framework/react-native@^0.4.0 @aries-framework/core@^0.4.0 @aries-framework/indy-sdk@^0.4.0 @aries-framework/anoncreds@^0.4.0 indy-sdk-react-native@^0.3.1
```

File: guides/updating/versions/0.3-to-0.4.md [hunk 3]
```diff
@@ -0,0 +1,875 @@
...
+In addition the `connectToIndyLedgersOnStartup` property has been removed in favour of a per-network `connectOnStartup` property that allows more fine-grained control over which networks to connect to on startup.
+
+<!--tabs-->
+
+##### 0.3.x
+
+```ts
+import { Agent } from '@aries-framework/core'
+
+// Import from @aries-framework/react-native in React Native
```
Comment (blu3beri) [inline guides/updating/versions/0.3-to-0.4.md:134]: Do we not want to do the tabs here?

File: guides/updating/versions/0.3-to-0.4.md [hunk 4]
```diff
@@ -0,0 +1,875 @@
...
+
+```ts
+import { Agent, DidsModule } from '@aries-framework/core'
+import { IndySdkModule } from '@aries-framework/indy-sdk'
+
+// Import from @aries-framework/react-native in React Native
+import { agentDependencies } from '@aries-framework/node'
+
+// Import from indy-sdk-react-native in React Native
+import indySdk from 'indy-sdk'
```
Comment (blu3beri) [inline guides/updating/versions/0.3-to-0.4.md:239]: indySdk does not seem to be used?

File: guides/updating/versions/0.3-to-0.4.md [hunk 5]
```diff
@@ -0,0 +1,875 @@
...
+    /* ... */
+  },
+  dependencies: agentDependencies,
+  modules: {
+    indySdk: new IndySdkModule({
+      indySdk,
+      networks: [
+        {
+          id: 'Sovrin Mainnet',
+          // Important: make sure to pick the correct indy namespace for the network you're connecting to
```
Comment (blu3beri) [inline guides/updating/versions/0.3-to-0.4.md:317]: Do we include a link somewhere on where to find it? Maybe the anoncreds registry list? Not sure if there are better resources for this.

File: guides/updating/versions/0.3-to-0.4.md [hunk 6]
```diff
@@ -0,0 +1,875 @@
...
+    issuerId: 'did:indy:sovrin:KEPUoXHcswsfsHkXGfmXB7',
+    name: 'Example Schema',
+    version: '1.0.0',
+  },
+  options: {},
+})
+
+// The resolver and registrar follow the same pattern as the Did Registration
+// and Did Resolver interfaces as used by the did module.
+if (schemaResult.schemaState.state !== 'finished') {
```
Comment (blu3beri) [inline guides/updating/versions/0.3-to-0.4.md:514]: Might be good, later on, to add a utility method for this like `assertFinished` or `assertState`. 

We would expose both then.

**Review by blu3beri — APPROVED** [review]
Awesome! It looks very well written and very extensive. 

One comment overall, do we not want to use the "dynamic" / type-checked snippets for this? Where we create ts files, and could also test them?

File: guides/updating/versions/0.3-to-0.4.md (hunk 3 above)
Comment (TimoGlastra) [reply guides/updating/versions/0.3-to-0.4.md:134]: We can't do tabs inside tabs :/

File: guides/updating/versions/0.3-to-0.4.md (hunk 5 above)
Comment (TimoGlastra) [reply guides/updating/versions/0.3-to-0.4.md:317]: There's not really a list. I'll open an issue in indy-did-networks and point towards it

File: guides/updating/versions/0.3-to-0.4.md (hunk 6 above)
Comment (TimoGlastra) [reply guides/updating/versions/0.3-to-0.4.md:514]: Yes that would be a good one to add!

**Comment by TimoGlastra** [comment]
> One comment overall, do we not want to use the "dynamic" / type-checked snippets for this? Where we create ts files, and could also test them?

I don't think that's worth it. The scripts are for a specific version, so it means we need to have all versions of AFJ installed, and it doesn't change over time.

**Pull request merged**